jaeger-core/src/main/java/com/uber/jaeger/baggage/BaggageSetter.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2017, Uber Technologies, Inc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.jaeger.baggage;
+
+import com.uber.jaeger.Span;
+import com.uber.jaeger.SpanContext;
+import com.uber.jaeger.metrics.Metrics;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.Value;
+
+/**
+ * BaggageSetter is a class that sets baggage and the logs associated
+ * with the baggage on a given {@link Span}.
+ */
+@Value(staticConstructor = "of")
+public class BaggageSetter {
+  final boolean valid;
+  final int maxValueLength;
+  final Metrics metrics;
```

Comment:
* to make fields visible to subclasses, they should be `protected`. Did you also need package visibility?
 * given that these fields are final, there should be a constructor setting them. If they are only set by subclasses, then the constructor can be protected (but then this class should be abstract).
 * the concept of `valid` is not clear, please document